protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -814,6 +815,38 @@ private long parseUnsignedInteger(String value) throws DecodingException {
         boolean isValid(char chr);
     }
 
+    /**
+     * Verifies subsequent characters match a specified string
+     */
+    public static class StringValidator implements CharacterValidator {
```

Comment:
does it make sense to say that StringValidator is a CharacterValidator?
By the way, Is matching a full string a validation?

Reply:
> does it make sense to say that StringValidator is a CharacterValidator?
> By the way, Is matching a full string a validation?

This is what the *master* code is doing, but with disperse class overload. Centralizing and testing such implemmentation makes sens to me.

Despite all its weaknesses, this `StringValidator` looks like a valuable refactoring step to me. If we want to go further on this topic we might need to address the way the `ImapRequestLineReader` is working, which seems like a complex task to me.

Reply:
It doesn't need further refactoring, it's about naming concepts.
A CharacterValidator takes a character and validate it.
A StringValidator should take a String and validate it. So your StringValidator is actually a CharacterValidator (because it validates characters) that tries to _match_ a given String.
I would call it a StringMatcherCharacterValidator or StringMatcher.